pkg/util/kubernetes/hpa/hpa_test.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package hpa
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func newMockConfigMap(metricName string, labels map[string]string) *v1.ConfigMap {
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      datadogHPAConfigMap,
+			Namespace: "default",
+		},
+	}
+	if metricName == "" || len(labels) == 0 {
+		return cm
+	}
+
+	custMetric := CustomExternalMetric{
+		Name:      metricName,
+		Labels:    labels,
+		Timestamp: 12,
+		HpaName:   "foo",
+		Value:     1,
+		Valid:     false,
+	}
+	cm.Data = make(map[string]string)
+	marsh, _ := json.Marshal(custMetric)
+	cm.Data[metricName] = string(marsh)
+	return cm
+}
+
+func newMockHPAExternalManifest(metricName string, labels map[string]string) *v2beta1.HorizontalPodAutoscaler {
+	return &v2beta1.HorizontalPodAutoscaler{
+		Spec: v2beta1.HorizontalPodAutoscalerSpec{
+			Metrics: []v2beta1.MetricSpec{
+				{
+					External: &v2beta1.ExternalMetricSource{
+						MetricName: metricName,
+						MetricSelector: &metav1.LabelSelector{
+							MatchLabels: labels,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestRemoveEntryFromConfigMap(t *testing.T) {
+	hpaCl := HPAWatcherClient{
+		clientSet: fake.NewSimpleClientset(),
+		ns:        "default",
+	}
+	_, err := hpaCl.clientSet.CoreV1().ConfigMaps("default").Get(datadogHPAConfigMap, metav1.GetOptions{})
+	fmt.Printf("err is %s \n", err)
+	assert.True(t, errors.IsNotFound(err))
+
+	testCases := []struct {
+		caseName          string
+		configmap         *v1.ConfigMap
+		hpa               *v2beta1.HorizontalPodAutoscaler
+		expectedConfigMap map[string]string
+	}{
+		{
+			caseName:          "Metric exists, deleting",
+			configmap:         newMockConfigMap("foo", map[string]string{"bar": "baz"}),
+			hpa:               newMockHPAExternalManifest("foo", map[string]string{"bar": "baz"}),
+			expectedConfigMap: map[string]string{},
+		},
+		{
+			caseName:          "Metric is not listed, no-op",
+			configmap:         newMockConfigMap("foobar", map[string]string{"bar": "baz"}),
+			hpa:               newMockHPAExternalManifest("foo", map[string]string{"bar": "baz"}),
+			expectedConfigMap: map[string]string{"foobar": "{\"name\":\"foobar\",\"labels\":{\"bar\":\"baz\"},\"ts\":12,\"hpa_name\":\"foo\",\"value\":1,\"valid\":false}"},
+		},
+	}
+
+	for i, testCase := range testCases {
+		t.Run(fmt.Sprintf("#%d %s", i, testCase.caseName), func(t *testing.T) {
+			hpaCl.clientSet.CoreV1().ConfigMaps("default").Create(testCase.configmap)
+
```

Comment:
Your configmaps have the same name and use the same fake client. If these tests are run in parallel then it could cause flakiness.

Reply:
let's add a card 👍